### ISSUE TITLE
Pulsar doesn't build on Windows 10 or on Mac with Maven 3.6.3

### ISSUE THREAD

**Issue by devinbost** [apache/pulsar#7749]
**Describe the bug**
Maven is reporting a successful build without actually building anything. 

**To Reproduce**
1. Clone Pulsar master
2. Run `mvn clean install` at project root
3. Maven downloads dependencies but does not produce build output
4. Again, run `mvn clean install`
5. The result is this:

> (base) C:\Users\devin\src\pulsar>mvn clean install
> [INFO] Scanning for projects...
> [INFO] ------------------------------------------------------------------------
> [INFO] Detecting the operating system and CPU architecture
> [INFO] ------------------------------------------------------------------------
> [INFO] os.detected.name: windows
> [INFO] os.detected.arch: x86_64
> [INFO] os.detected.classifier: windows-x86_64
> [INFO]
> [INFO] ----------------------< org.apache.pulsar:pulsar >----------------------
> [INFO] Building Pulsar 2.7.0-SNAPSHOT
> [INFO] --------------------------------[ pom ]---------------------------------
> [INFO]
> [INFO] --- maven-clean-plugin:2.4.1:clean (default-clean) @ pulsar ---
> [INFO] Deleting C:\Users\devin\src\pulsar\target
> [INFO]
> [INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-maven) @ pulsar ---
> [INFO]
> [INFO] --- git-commit-id-plugin:3.0.0:revision (git-info) @ pulsar ---
> [INFO]
> [INFO] --- maven-remote-resources-plugin:1.5:process (process-resource-bundles) @ pulsar ---
> [INFO]
> [INFO] --- maven-site-plugin:3.5.1:attach-descriptor (attach-descriptor) @ pulsar ---
> [INFO]
> [INFO] --- maven-install-plugin:2.5.2:install (default-install) @ pulsar ---
> [INFO] Installing C:\Users\devin\src\pulsar\pom.xml to C:\Users\devin\.m2\repository\org\apache\pulsar\pulsar\2.7.0-SNAPSHOT\pulsar-2.7.0-SNAPSHOT.pom
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD SUCCESS
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time:  10.486 s
> [INFO] Finished at: 2020-08-04T13:45:21-06:00
> [INFO] ------------------------------------------------------------------------

Hence, the build succeeds but gives no output and doesn't run tests.

**Comment by aahmed-se** [comment]
We don't work with maven 3.6.3 ,  try 3.6.1

**Comment by murong00** [comment]
As the modules moved to profiles, you should tick "main" or "core-modules" checkbox in Maven panel.

**Comment by BewareMyPower** [comment]
Because the modules have been moved to profile section after #7566 , see the `pom.xml` in the project directory:

```xml
    <profile>
      <id>core-modules</id>
      <modules>
        <module>buildtools</module>
        <!-- more modules... -->
```

So you need run `mvn install -Pcore-modules` if you only want to install the modules of `core-modules` profile. For indices generation of Intellj Idea, you should click the associated checkbox in profile to choose the modules:

![](https://ftp.bmp.ovh/imgs/2020/08/64da8a1d29239e64.png)

**Comment by devinbost** [comment]
Thanks for the help on this.
I rolled back to Maven 3.6.0, and I was able to build Pulsar successfully. 
However, I'm having a new problem now which is that I'm having trouble debugging the function for my PR here: https://github.com/apache/pulsar/pull/7704

When I set a breakpoint in IntelliJ and run the tests from the Maven runner, it runs the tests, but my breakpoint is never hit, even if I run the tests in debug mode like this:
![image](https://user-images.githubusercontent.com/7418031/89466428-13932080-d731-11ea-86c7-2f4e678c11df.png)

If I try to run the test as I would in the past, like this:
![image](https://user-images.githubusercontent.com/7418031/89466527-3ae9ed80-d731-11ea-924f-7fb61c576827.png)

then I get this build error:

![image](https://user-images.githubusercontent.com/7418031/89466549-476e4600-d731-11ea-9dff-6debc039aa89.png)

So, how do I debug my test now?
